analysis/__init__.py

```diff
@@ -13,18 +13,24 @@ def get_file_path():
 from utils import get_people_mentioned_new
 
 
-def get_article_info(article_text, ann=None, verbose=False):
+def get_article_info(article_text, ann=None, verbose=False, hostname=None, port=None):
     """
     Given a piece of text, runs it through our entire pipeline.
 
     Can optionally pass it the CoreNLP annotation if it was precomputed.
     """
     if ann is None:
+        kwargs = {}
+        if hostname:
```

Comment:
Looking at how `hostname` is used in `nlp_utils.annotate_corenlp`, it looks like it will use the OS environment variable if it is set, regardless of `hostname`.

One possible fix is to change `nlp_utils.annotate_corenlp` so that it:
- uses `hostname` if `hostname` is set to something other than `localhost`
- uses the environment variables if `hostname` is set to `localhost`
- uses `localhost` if the environment variables are not set

Although it's a bit messy, I feel some messiness is inevitable since we have multiple default values. We do need to specify a pecking order though.

I feel that both `annotate_corenlp` and `get_article_info` should ideally have the same default arguments for hostname and port. (Or if you know that `annotate_corenlp` is only called from `get_article_info`, maybe remove the defaults from `annotate_corenlp`?)

Reply:
Yeah I like your sugggestion. The priority order of explicitly passed argument > environment > default makes sense. And yeah, let me remove the default arguments from `annotate_corenlp`, that makes sense to me.